src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,73 @@
 package core.basesyntax;
 
+import static java.lang.Integer.valueOf;
+
 public class SalaryInfo {
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+
+        // defence against empty input
+        if (names == null
+                || data == null
+                || dateFrom == null
+                || dateTo == null) {
+            throw new RuntimeException("Empty data");
+        }
+
+        //array with int = 0 for each name, which will be used to sum up salaries
+        int [] results = new int[names.length];
+
+        // both dates converted into numbers yyyymmdd
+        String[] dateFromSplitted = dateFrom.split("\\.");
+        int dateFromConvertedIntoNumber = valueOf(dateFromSplitted[2]) * 10000
+                + valueOf(dateFromSplitted[1]) * 100
+                + valueOf(dateFromSplitted[0]);
+
+        String [] dateToSplitted = dateTo.split("\\.");
+        int dateToConvertedIntoNumber = valueOf(dateToSplitted[2]) * 10000
+                + valueOf(dateToSplitted[1]) * 100
+                + valueOf(dateToSplitted[0]);
```

Comment:
Use LocalDate instead of Date. It's more modern and provides a cleaner API for dealing with dates.